src/core/constraint.jl

```diff
@@ -40,12 +40,12 @@ function constraint_restore_all_items(pm, n)
     z_branch = _PMs.var(pm, n, :z_branch)
     z_bus = _PMs.var(pm, n, :z_bus)
 
-    for (i,load) in  _PMs.ref(pm, n, :load)
-        JuMP.@constraint(pm.model, z_demand[i] == 1)
-    end
-    for (i,shunt) in  _PMs.ref(pm, n, :shunt)
-        JuMP.@constraint(pm.model, z_shunt[i] == 1)
-    end
+    # for (i,load) in  _PMs.ref(pm, n, :load)
+    #     JuMP.@constraint(pm.model, z_demand[i] == 1)
+    # end
+    # for (i,shunt) in  _PMs.ref(pm, n, :shunt)
+    #     JuMP.@constraint(pm.model, z_shunt[i] == 1)
+    # end
```

Comment:
If the network cannot fully support the load (i.e. some devices have status=0 and there is not enough generation to support the full load) then the restoration sequence is infeasible.  

Alternatively, this can also provide a restoration sequence that can support the entire load, but the objective function balances voltage violations and will not provide full load.  

I removed it because restoration is infeasible when picking a subset of devices for restoration.


I'm open to discussion on whether this is the appropriate choice.

Reply:
Seems like a reasonable rational, but I suppose it depends on the goals of this routine.  Another options would be to only support networks where the full load can be supported a the end of the restoration and set the status of some loads to zero before solving the case, if you are shooting for a partial restoration.  Can you compile a list of the types of cases you are envisioning supporting and what the core assumptions are?  (e.g. full vs partial restoration, fixed dispatch vs flexible dispatch, ...)

Reply:
Constraint restored to require full load. A seperate constraint function (inside of `iterative_restoration.jl`) only requires devices to be restored and allows partial load restoration.